celements-model/src/main/java/com/celements/store/part/CelHibernateStoreDocumentPart.java

```diff
@@ -359,9 +371,19 @@ public XWikiDocument loadXWikiDoc(XWikiDocument doc, XWikiContext context) throw
     return doc;
   }
 
+  @SuppressWarnings("unchecked")
+  private ImmutableMap<String, BaseObject> loadExistingXObjects(XWikiDocument doc,
+      XWikiContext context) {
+    Query query = store.getSession(context).createQuery(
```

Comment:
Why sending a query to the database instead of using the complete object list in document cache?